src/libraries/Common/src/System/Net/CookieComparer.cs

```diff
@@ -5,23 +5,29 @@ namespace System.Net
 {
     internal static class CookieComparer
     {
-        internal static int Compare(Cookie left, Cookie right)
+        internal static bool Equals(Cookie left, Cookie right)
         {
-            int result;
-
-            if ((result = string.Compare(left.Name, right.Name, StringComparison.OrdinalIgnoreCase)) != 0)
+            if (!string.Equals(left.Name, right.Name, StringComparison.OrdinalIgnoreCase))
             {
-                return result;
+                return false;
             }
 
-            if ((result = string.Compare(left.Domain, right.Domain, StringComparison.OrdinalIgnoreCase)) != 0)
+            if (!EqualsDomain(left.Domain, right.Domain))
             {
-                return result;
+                return false;
             }
 
             // NB: Only the path is case sensitive as per spec. However, many Windows applications assume
             //     case-insensitivity.
-            return string.Compare(left.Path, right.Path, StringComparison.Ordinal);
+            return string.Equals(left.Path, right.Path, StringComparison.Ordinal);
+        }
+
+        internal static bool EqualsDomain(ReadOnlySpan<char> left, ReadOnlySpan<char> right)
+        {
+            if (left.StartsWith(".")) left = left.Slice(1);
+            if (right.StartsWith(".")) right = right.Slice(1);
```

Comment:
It may not matter, but I expect it'd be a bit faster to do:
```
if (!left.IsEmpty && left[0] == '.')
```

Reply:
Done